docs/commits_and_prs.md

```diff
@@ -0,0 +1,105 @@
+Writing Commits and Pull Requests
+=================================
+
+Small, atomic commits and pull requests (PRs) with helpful commit messages,
+PR titles and PR bodies are great for collaboration and key to a maintainable
+project. Tools like [`git rebase`](https://docs.github.com/en/get-started/using-git/using-git-rebase-on-the-command-line)
+can help you to rewrite commits and split up branches to tell a clean and coherent story.
+
+Make small PRs and atomic commits
+---------------------------------
+
+A good guide is that each PR should do "one thing": a single logical change.
+Needing "and" in your PR title might be a sign that you should split it up. If
+you discover a bug while adding a new feature, send a separate PR to
+fix the bug. If code doesn't follow current conventions do a refactoring commit
+first then a separate commit to make your functional change. We like small PRs
+because:
+
+* They're necessary for [continuous deployment](merging.md): you can't deploy a small diff without first merging a small PR.
+* They get faster, better reviews: it takes less time and context-switching to review a small, focused PR.
+* They improve the quality of your work by encouraging clearly organized thinking, modular code, and purity of purpose.
+
+Make commits atomic: CI should be passing after each commit to enable tools
+like `git revert` and `git bisect` to be used effectively and to make history
+easier to browse and search by avoiding "fix", "typo" and "test" commits. Run
+`make sure` locally to make sure that CI will pass.
+
+<details><summary><b>When should you include more than one commit in a PR?</b></summary>
+If a few closely related commits would be easier to review and deploy together
+then you might want to include them in the same PR.
+If commits can be decoupled into independently-reviewable, smaller PRs then it
+might be better to split them up.
+</details>
+
+Write helpful commit messages, PR titles, and PR bodies
```

Comment:
Sticking with body because that's what GitHub calls it (for example GitHub CLI: the `--body` argument to `gh pr create|edit` or `gh pr view --json body`, It's body in the GitHub REST API too) but I think it's now consistent about commit "messages" but PR "titles and bodies"